lib/timeutils/tests/test_wallclocktime.c

```diff
@@ -49,10 +49,31 @@ Test(wallclocktime, test_strptime_parses_broken_down_time)
   cr_expect(wct.wct_hour == 18);
   cr_expect(wct.wct_min == 23);
   cr_expect(wct.wct_sec == 12);
+  cr_expect(wct.wct_usec == 12345);
 
   cr_expect(wct.wct_gmtoff == -1);
 }
 
+Test(wallclocktime, test_strptime_parses_truncated_usec)
+{
+  WallClockTime wct = WALL_CLOCK_TIME_INIT;
+  gchar *end;
+
+  end = wall_clock_time_strptime(&wct, "%b %d %Y %H:%M:%S.%f", "Jan 16 2019 18:23:12.012X");
+  cr_assert(*end == 'X');
+  cr_expect(wct.wct_usec == 12000);
+}
+
+Test(wallclocktime, test_strptime_parses_overflowed_usec)
+{
+  WallClockTime wct = WALL_CLOCK_TIME_INIT;
+  gchar *end;
+
+  end = wall_clock_time_strptime(&wct, "%b %d %Y %H:%M:%S.%f", "Jan 16 2019 18:23:12.0123456");
+  cr_assert(*end == '6');
+  cr_expect(wct.wct_usec == 12345);
+}
```

Comment:
I think in this case we should consume the numbers but ignore them.

Reply:
Makes sense, I'll work on this.